File: Source/OldCode/Archive/2005 TVA Code Library/Version/CS/TVA.Services_CS/ClientInfo.cs
//*******************************************************************************************************
//  ClientInfo.cs
//  Copyright © 2008 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: Pinal C. Patel
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-3024
//       Email: pcpatel@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  09/12/2006 - Pinal C. Patel
//       Generated original version of source code.
//  09/30/2008 - James R Carroll
//       Converted to C#.
//
//*******************************************************************************************************

using System;
using System.Threading;
using System.Collections.Generic;

namespace TVA.Services
{
	[Serializable()]
    public class ClientInfo
	{
        #region [ Members ]

        // Fields
        public Guid ClientID;
        public ApplicationType ClientType;
        public string ClientName;
        public string UserName;
        public string MachineName;
        public DateTime ConnectedAt;

        #endregion

        #region [ Constructors ]

        public ClientInfo(Guid clientID)
        {
            ClientID = clientID;
            ClientType = Common.GetApplicationType();
            
            UserName = Thread.CurrentPrincipal.Identity.Name;

            if (string.IsNullOrEmpty(UserName))
                UserName = Environment.UserDomainName + "\\" + Environment.UserName;

            MachineName = Environment.MachineName;

            if (ClientType == ApplicationType.WindowsCui || ClientType == ApplicationType.WindowsGui)
            {
                ClientName = AppDomain.CurrentDomain.FriendlyName;
            }
            else if (ClientType == ApplicationType.Web)
            {
                ClientName = System.Web.HttpContext.Current.Request.ApplicationPath;
            }
        }

        #endregion
	}
}

File: Source/OldCode/Archive/2005 TVA Code Library/Version/3.0/Local/Source/TVA.Services/ClientInfo.vb
' 09-12-06

Imports TVA.Assembly

<Serializable()> _
Public Class ClientInfo

    Public Sub New(ByVal clientID As Guid)

        MyBase.New()
        Me.ClientID = clientID
        ClientType = TVA.Common.GetApplicationType()
        UserName = System.Threading.Thread.CurrentPrincipal.Identity.Name
        If String.IsNullOrEmpty(UserName) Then
            UserName = Environment.UserDomainName & "\" & Environment.UserName
        End If
        MachineName = Environment.MachineName
        Select Case ClientType
            Case ApplicationType.WindowsCui, ApplicationType.WindowsGui
                ClientName = AppDomain.CurrentDomain.FriendlyName
            Case ApplicationType.Web
                ClientName = System.Web.HttpContext.Current.Request.ApplicationPath
        End Select

    End Sub

    Public ClientID As Guid
    Public ClientType As ApplicationType
    Public ClientName As String
    Public UserName As String
    Public MachineName As String
    Public ConnectedAt As Date

End Class
